### ISSUE TITLE
Feature request: Generic way to allow proving ownership of a username on an arbitrary domain

### ISSUE THREAD

**Issue by sariyamelody** [keybase/keybase-issues#813]
A framework that websites can implement for verification of a username on their site would be **awesome**.

**Comment by kojiromike** [comment]
I bumped into this issue trying to figure out how to get KeyBase to prove my StackOverflow account. Seems old, but it's not closed so…

Related: [How to link my stackoverflow profile with arbitrary third party identity sites](http://meta.stackoverflow.com/questions/314749/how-to-link-my-stackoverflow-profile-with-arbitrary-third-party-identity-sites).

**Comment by timmc** [comment]
At this point I feel like working up something for the generic case is probably easier than going through all of https://github.com/keybase/keybase-issues/issues/518 individually. :-P

**Comment by jlb1234** [comment]
@timmc I think that's the actual solution. There are just too many places where you *should* be able to prove your identity but there isn't going to be a move to do them all. I put together some thoughts at #2052

**Comment by timmc** [comment]
A challenge: A generic web user proof would not have protection against homoglyph attacks. Currently Keybase-the-service vets all new services to make sure there aren't homoglyph attacks, but a self-service method would not permit that.

**Comment by jlb1234** [comment]
What exactly is a homoglyph attack? Is that where two usernames could be different but look the same due to Unicode?

**Comment by timmc** [comment]
Yes, exactly. The worst you can do with just ASCII is `1lI` or `O0` but having a good font is usually enough to counter that, and casefolding everything helps a bit too: `1li o0`. If keybase were to allow arbitrary sites, it would have to restrict to lowercase printable ASCII, at least at first.
